WordPressAuthenticatorTests/GoogleSignIn/GoogleOAuthTokenGettingStub.swift

```diff
@@ -0,0 +1,25 @@
+@testable import WordPressAuthenticator
+
+struct GoogleOAuthTokenGettingStub: GoogleOAuthTokenGetting {
+
+    let result: Result<OAuthTokenResponseBody, Error>
+
+    init(response: OAuthTokenResponseBody) {
+        self.init(result: .success(response))
+    }
+
+    init(error: Error) {
+        self.init(result: .failure(error))
+    }
+
+    init(result: Result<OAuthTokenResponseBody, Error>) {
+        self.result = result
+    }
+
+    func getToken(clientId: GoogleClientId, authCode: String, pkce: ProofKeyForCodeExchange) async throws -> OAuthTokenResponseBody {
+        switch result {
+        case .success(let response): return response
+        case .failure(let error): throw error
+        }
+    }
+}
```

Comment:
This test double is quite similar to the one for `DataGetting`. Both use a `Result<T, Error>` under the hood to stub a `throws -> T` method. 

If I'll need to create another `throws -> T` types, I might spend some time to write an abstraction to use to create more of these.